devp2p/discovery.py

```diff
@@ -218,7 +222,7 @@ def __init__(self, app, transport):
 
     def get_node(self, nodeid, address=None):
         "return node or create new, update address if supplied"
-        assert isinstance(nodeid, str)
+        assert isinstance(nodeid, bytes)
```

Comment:
Since `str == bytes` in Python 2, I guess this is want we want. What error condition do you have in mind?